### PR TITLE
Add an example for X.509 path validation

### PR DESCRIPTION
**Pull request by securitykernel**
Adds an example for simple X.509 path validation of an end entity certificate against a Root CA.

### CODE REVIEW

File: src/examples/x509_path.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+   Botan::Path_Validation_Restrictions restrictions;
+
+   // Optional: Specify usage type, compared against the key usage in endEntityCert
+   Botan::Usage_Type usage = Botan::Usage_Type::UNSPECIFIED;
+
+   // Optional: Specify hostname, if not empty, compared against the DNS name in endEntityCert
+   std::string hostname = "";
+
+   Botan::Path_Validation_Result validationResult =
+      Botan::x509_path_validate(end_entity, restrictions, store, hostname, usage);
```
Comment (reneme) [inline src/examples/x509_path.cpp:22]: Perhaps use the overload that takes a list of certificates for `end_entity` to showcase the path building capability as well. Also, there's another overload that takes a list of `Certificate_Store*`, with that you could showcase the `System_Certificate_Store` as well; basically validating against "anything the OS trusts AND this application specific root".

**Review by reneme — APPROVED** [review]
Kudos for working on the documentation!

File: src/examples/x509_path.cpp (hunk 1 above)
Comment (securitykernel) [reply src/examples/x509_path.cpp:22]: Good point, I'll change it.

Comment (securitykernel) [reply src/examples/x509_path.cpp]: Done

**Comment by randombit** [comment]
:+1:

**Pull request merged**